client/student/app/lecture/comment.form.directive.js

```diff
@@ -35,25 +35,38 @@
         
         function submitComment() {
             commentForm.submitted = true;
-            lectureFactory.submitComment(commentForm.comment, function(response) {
-                console.log('Success');
-                commentForm.feedbackMessage = 'Comment submitted!';
-                commentForm.feedbackType = 'alert-success';
-                commentForm.feedbackIcon = 'glyphicon-ok-sign';
-                
-                var comment = {
-                    id: response.id,
-                    content: commentForm.comment,
-                    submissionTime: new Date()
-                };
-                
-                commentForm.list.push(comment);
-            }, function() {
+            
+            var commentMaxLength = 500;
```

Comment:
I would prefer if the initialization is moved to the top as 
```

var commentForm = this;
commentForm.submitComment = submitComment;
commentForm.maxLength = 500;

```

This way it can be bound to the html like this

```

<input type="text" 
             class="form-control" 
             placeholder="Comment..." 
             data-ng-model="commentForm.comment"
             maxlength="{{commentForm.maxLength}}">

```

This should prevent the input accepting more than 500 characters.. which would mean that the feedback want show :(

Angular has actually their own ng-maxlength attribute.
I have not tested it, but there is a demo at the bottom of

https://docs.angularjs.org/api/ng/directive/input

which gives an example of how you can show the feedback dynamically 
as soon as the input length reach 501.